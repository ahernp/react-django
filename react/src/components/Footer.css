#footer {
    display: block;
    background-color: #fff;
    position: fixed;
    bottom: 0;
    height: 45px;
    width: 100%;
    border-top: 1px #ccc solid;
    margin: 0;
    padding-left: 8px;
    z-index: 9;
}
