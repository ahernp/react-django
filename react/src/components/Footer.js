import React from 'react';

import './Footer.css';

import { ADMIN_URL } from '../constants'

const DEFAULT_FOOTER = (<React.Fragment>Dynamic Content <a key="admin-link" href={ADMIN_URL}>Admin</a></React.Fragment>);

export default class Footer extends React.Component {
    render() {
        return (
            <div id="footer">
                <p>{this.props.children ? this.props.children : DEFAULT_FOOTER}</p>
            </div>
        );
    }
}
