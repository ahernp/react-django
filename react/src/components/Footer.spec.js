import Footer from "./Footer";

it("renders correctly", () => {
  const wrapper = shallow(
    <Footer />
  );

  expect(wrapper).toMatchSnapshot();
});

it("renders its children", () => {
  const wrapper = shallow(
    <Footer>
        Hello Mum
    </Footer>
  );

  expect(wrapper).toMatchSnapshot();
});
