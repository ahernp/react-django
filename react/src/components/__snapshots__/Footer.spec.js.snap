// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="footer"
>
  <p>
    <React.Fragment>
      Dynamic Content 
      <a
        href="/admin/"
        key="admin-link"
      >
        Admin
      </a>
    </React.Fragment>
  </p>
</div>
`;

exports[`renders its children 1`] = `
<div
  id="footer"
>
  <p>
    Hello Mum
  </p>
</div>
`;
